x-pack/plugins/elastic_assistant/server/lib/langchain/graphs/default_assistant_graph/helpers.ts

```diff
@@ -91,27 +92,35 @@ export const streamGraph = async ({
       if (done) return;
 
       const event = value;
-      if (event.event === 'on_llm_stream') {
-        const chunk = event.data?.chunk;
-        // TODO: For Bedrock streaming support, override `handleLLMNewToken` in callbacks,
-        // TODO: or maybe we can update ActionsClientSimpleChatModel to handle this `on_llm_stream` event
-        if (event.name === 'ActionsClientChatOpenAI') {
-          const msg = chunk.message;
-
-          if (msg.tool_call_chunks && msg.tool_call_chunks.length > 0) {
-            /* empty */
-          } else if (!didEnd) {
-            if (msg.response_metadata.finish_reason === 'stop') {
-              handleStreamEnd(finalMessage);
-            } else {
-              push({ payload: msg.content, type: 'content' });
-              finalMessage += msg.content;
+      // only process events that are part of the agent run
+      if ((event.tags || []).includes(AGENT_NODE_TAG)) {
+        if (event.event === 'on_llm_stream') {
+          const chunk = event.data?.chunk;
+          // TODO: For Bedrock streaming support, override `handleLLMNewToken` in callbacks,
+          // TODO: or maybe we can update ActionsClientSimpleChatModel to handle this `on_llm_stream` event
```

Comment:
FYI, this might be resolved with Patryk's work on BedrockChat: https://github.com/elastic/kibana/pull/186809. Will still be needed for `SimpleChatModel` though